services/data/db_utils.py

```diff
@@ -67,11 +67,28 @@ def get_latest_attempt_id_for_tasks(artifacts):
 
 
 def filter_artifacts_for_latest_attempt(artifacts):
+    """
+    filters artifacts of latest task attempt. 
+    Args:
+        artifacts (List[ArtifactRow]): a dictionary form of `ArtifactRow`
+
+    Returns:
+        `list` of filtered artifacts for the latest attempt
+    """
     attempt_ids = get_latest_attempt_id_for_tasks(artifacts)
     return filter_artifacts_by_attempt_id_for_tasks(artifacts, attempt_ids)
 
 
 def filter_artifacts_by_attempt_id_for_tasks(artifacts, attempt_for_tasks):
+    """
+    filters artifacts based on the attempt_ids. 
+    Args:
+        artifacts (List[ArtifactRow]): a dictionary form of `ArtifactRow`
+        attempt_for_tasks ([dict]): dictionary of {task_id:attempt_id}
+
+    Returns:
+        `list` of artifacts filtered based on `attempt_id`
+    """
```

Comment:
Eh you could as well use type annotations, they are already used pretty widely in `metaflow-service` codebase. And I don't think this style of docstring is used anywhere else here.